src/workspaces/getWorkspaces/rushWorkspaces.ts

```diff
@@ -0,0 +1,31 @@
+import findUp from "find-up";
+import path from "path";
+import { RushConfiguration } from "@microsoft/rush-lib";
+
+import { WorkspaceInfo } from "../../types/WorkspaceInfo";
+
+export function getRushWorkspaces(cwd: string): WorkspaceInfo {
+  try {
+    const rushJsonPath = findUp.sync("rush.json", { cwd });
+    if (!rushJsonPath) {
+      return [];
+    }
+
+    const rushConfig = RushConfiguration.loadFromConfigurationFile(
+      rushJsonPath
+    );
+
+    return rushConfig.projects.map((project) => {
+      return {
+        name: project.packageName,
+        path: project.projectFolder,
+        packageJson: {
+          ...project.packageJson,
+          packageJsonPath: path.join(project.projectFolder, "package.json"),
```

Comment:
I don't know if this will always be true? Maybe Rush has this information embedded somewhere else?

Reply:
The projectfolder from rush points to the root of a package directly - it's not a minimatch. It should hold that rush projects have a package.json at its root.